server/server.js

```diff
@@ -0,0 +1,264 @@
+// @flow
+import type {IncomingMessage, ServerResponse} from 'http'
+import type {NextConfig} from './config'
+
+import {STATUS_CODES} from 'http'
+import {parse as parseUrl} from 'url'
+import {parse as parseQuerystring} from 'querystring'
+import {resolve, join, sep} from 'path'
+import send from 'send'
+import {PHASE_PRODUCTION_SERVER, PHASE_DEVELOPMENT_SERVER} from '../lib/constants'
+import getConfig from './config'
+import * as asset from '../lib/asset' // can be deprecated
+
+type ServerConstructor = {
+  dir?: string,
+  dev?: boolean,
+  quiet?: boolean,
+  conf?: NextConfig
+}
+
+type UrlParseResult = {
+  protocol?: string;
+  slashes?: boolean;
+  auth?: string;
+  host?: string;
+  port?: string;
+  hostname?: string;
+  hash?: string;
+  search?: string;
+  query?: any; // null | string | Object
+  pathname?: string;
+  path?: string;
+  href: string;
+}
+
+type ParsedUrl = {
+  protocol?: string;
+  slashes?: boolean;
+  auth?: string;
+  host?: string;
+  port?: string;
+  hostname?: string;
+  hash?: string;
+  search?: string;
+  query: any; // null | string | Object
+  pathname: string;
+  path: string;
+  href: string;
+}
+
+type Envelope = {
```

Comment:
You probably need to `export` these?

Reply:
yeah eventually, it's a work in progress 😄